### PR TITLE
feat(helm): update chart with loki version 3.2.0

### PR DESCRIPTION
**Pull request by Starefossen**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: production/helm/loki/Chart.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ apiVersion: v2
 name: loki
 description: Helm chart for Grafana Loki and Grafana Enterprise Logs supporting both simple, scalable and distributed modes.
 type: application
-appVersion: 3.1.1
-version: 6.14.1
+appVersion: 3.2.0
+version: 6.15.0
```
Comment (DylanGuedes) [inline production/helm/loki/Chart.yaml:6]: I think you'll need to specify a new version for this helm chart (like 6.16)

**Review by DylanGuedes — CHANGES_REQUESTED** [review]
thanks for your review! we'll need to release a new version with this, we can't leave it pointing to an already released version

File: production/helm/loki/Chart.yaml (hunk 1 above)
Comment (Starefossen) [reply production/helm/loki/Chart.yaml]: Good catch 😅

**Comment by Starefossen** [comment]
@DylanGuedes is this good to go?

**Comment by trevorwhitney** [comment]
Is this an issue with forks not having access to GitHub Action secrets? `app-id` is provided from a secret, and it looks like it's passing from branches but not forks?

**Review by DylanGuedes — APPROVED** [review]

**Comment by DylanGuedes** [comment]
> @DylanGuedes is this good to go?

sorry, it is/was, but because we recently changed how our CI does the package publishing, we'll need you to rebase one more time and to also update the version from 6.16.0 to 6.17.0 (we had to do a release in between and we preferred it to be before using loki 3.2.0 for stability reasons)

**Review by DylanGuedes — APPROVED** [review]

**Comment by DylanGuedes** [comment]
hey I believe this [lint step](https://github.com/grafana/loki/actions/runs/11105484410/job/30851746772?pr=14281) keeps breaking because rebasing isn't updating your branch code to receive the changes since they're for a different folder. 
super annoying but do you mind copying `.github/workflows/helm-ci.yaml` from main branch to your branch and pushing a commit with it?

**Comment by Starefossen** [comment]
@trevorwhitney any reason why I can not find this version up on Artifacthub? https://artifacthub.io/packages/helm/grafana/loki

**Comment by trevorwhitney** [comment]
@Starefossen we had an issue with our release automation, you should be able to find version `6.18` of the chart now, and it has Loki 3.2

**Comment by Starefossen** [comment]
Thanks for fixing it!

**Pull request merged**